### DISCUSSION TITLE
Autocompletion with class `__call__()` doesn't work

### DISCUSSION THREAD

**Discussion by 4l1fe** [fastapi/typer#623] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#!/usr/bin/python3
# -*- coding: utf-8 -*-
from typer import Typer, Argument


cli = Typer()


class TargetNamesEnum(str, Enum):
    ser = 'service'


class Autocompletion:
    ARG_TARGET = 'service'

    def __init__(self, argument):
        elif argument == self.ARG_TARGET:
            self.autocomplete_enum = TargetNamesEnum

    def __call__(self, incomplete: str, *args, **kwargs):
        completion = []

        for name in self.autocomplete_enum.__members__.values():
            if name.startswith(incomplete):
                completion.append(name)

        return completion


@cli.command()
def run(service: TargetNamesEnum = Argument(None, show_default=False, autocompletion=Autocompletion('service'))


if __name__ == '__main__':
    cli()
```

### Description

If i try to get an argument completed in cli  with `Tab` i see the error.
```shell
$ script.py <Tab>

...
TypeError: <__main__.Autocompletion object at 0x7fe61088ac80> is not a module, class, method, or function.
```

While in a repl i can call well `Autocompletion` manually
```python
In [1]: import Autocompletion
In [2]: a = Autocompletion('service')
In [3]: a('')
Out[3]:
[<TargetNamesEnum.ser: 'service'>]
```

### Operating System

Linux

### Typer Version

0.7.0

### Python Version

3.10.6

**Comment by YuriiMotov** [comment]
Works if specify `__call__` method explicitly:
```
def run(
    service: TargetNamesEnum = Argument(
        None,
        show_default=False,
        autocompletion=Autocompletion('service').__call__
    )
):
    pass
```
